docs/architecture/adr-004-genesis-modes.md

```diff
@@ -0,0 +1,295 @@
+# ADR 004: Genesis Modes
+
+## Changelog
+
+- Sept 3, 2019: Initial draft
+
+## Context
```

Comment:
@fedekunze I synced with @rigelrozanski and updated the context. Let me know if this is inline with your thinking.